desktop_gui.py

```diff
@@ -85,28 +84,52 @@ def initUI(self):
         self.user_input = QLineEdit()
         user_input.addWidget(self.user_input)
         # add a branch button to the horizontal layout to branch the conversation
-        self.branch_button = QPushButton('')
-        # use an icon for the branch button
-        # obtained from https://www.iconsdb.com/black-icons/fork-2-icon.html 
-        # licensed under the MIT license
-        self.branch_button.setIcon(QIcon('branch.png'))
-        # connect the branch button to the create branch function in this file and pass the conversation id and the selected item
-        self.branch_button.clicked.connect(lambda: self.create_branch( self.conversation.currentText(), selected_item))
-        user_input.addWidget(self.branch_button)
+        # check if the config file has branching enabled
+        if branching_enabled == True:
+            self.branch_button = QPushButton('')
+            # use an icon for the branch button
+            # obtained from https://www.iconsdb.com/black-icons/fork-2-icon.html 
+            # licensed under the MIT license
+            self.branch_button.setIcon(QIcon('branch.png'))
+            # connect the branch button to the create branch function in this file and pass the conversation id  the selected item and the selected item id
+            # the selected item id is the position of the selected item in the tree view. ie the index of the selected item -1
+            self.branch_button.clicked.connect(lambda: self.create_branch(conversation_id, self.chat_history.selectedIndexes()[0].row() + 1))
+            user_input.addWidget(self.branch_button)
+
 
         # add a dropdown menu to the horizontal layout to select the conversation
-        self.conversation = QComboBox()
-        # call the set dropdown function in this file
-        self.set_dropdown()
+        # check if the config file has conversations enabled
+        if conversations_enabled == True:
+            self.conversation = QComboBox()
+            # call the set dropdown function in this file
+            self.set_dropdown()
+        
+            user_input.addWidget(self.conversation)
+            # connect the signal of the dropdown menu to the switch conversation function
+            self.conversation.currentTextChanged.connect(self.switch_conversation)
+        else:
+            # if conversations are not enabled, but history is enabled, we need to set the conversation id to the first conversation id
```

Comment:
I'm not sure how we want to handle this case, does it make sense to have history enabled but conversations not? What history is it supposed to remember?

Reply:
conversation is for whether of not there is a dropdown. history is for whether or not the front end remembers past conversational turns. in the case of conversation being off but history remaining one i currently have it set to use id 0. it was the easiest to impliment at the time